conda/cli/main_config.py

```diff
@@ -235,215 +224,22 @@ def execute_config(args, parser):
-                    print("--add", key, repr(item))
-
-
-    # PyYaml does not support round tripping, so if we use yaml.dump, it
-    # will clear all comments and structure from the configuration file.
-    # There are no yaml parsers that do this.  Our best bet is to do a
-    # simple parsing of the file ourselves.  We can check the result at
-    # the end to see if we did it right.
-
-    # First, do it the pyyaml way
-    new_rc_config = deepcopy(rc_config)
-
-    # Add
-    for key, item in args.add:
-        if key not in config.rc_list_keys:
-            common.error_and_exit("key must be one of %s, not %r" %
-                                  (config.rc_list_keys, key), json=args.json,
-                                  error_type="ValueError")
-        if not isinstance(rc_config.get(key, []), list):
-            raise CouldntParse("key %r should be a list, not %s." % (key,
-                rc_config[key].__class__.__name__))
-        if item in rc_config.get(key, []):
-            # Right now, all list keys should not contain duplicates
-            message = "Skipping %s: %s, item already exists" % (key, item)
-            if not args.json:
-                print(message, file=sys.stderr)
-            else:
-                json_warnings.append(message)
-            continue
-        new_rc_config.setdefault(key, []).insert(0, item)
-
-    # Set
-    for key, item in args.set:
-        yamlitem = yaml.load(item)
-        if not isinstance(yamlitem, bool):
-            common.error_and_exit("%r is not a boolean" % item, json=args.json,
-                                  error_type="TypeError")
-
-        new_rc_config[key] = yamlitem
-
-    # Remove
-    for key, item in args.remove:
-        if key not in new_rc_config:
-            common.error_and_exit("key %r is not in the config file" % key, json=args.json,
-                                  error_type="KeyError")
-        if item not in new_rc_config[key]:
-            common.error_and_exit("%r is not in the %r key of the config file" %
-                                  (item, key), json=args.json, error_type="KeyError")
-        new_rc_config[key] = [i for i in new_rc_config[key] if i != item]
-
-    # Remove Key
-    for key, in args.remove_key:
-        if key not in new_rc_config:
-            common.error_and_exit("key %r is not in the config file" % key, json=args.json,
-                                  error_type="KeyError")
-        del new_rc_config[key]
-
-    if args.force:
-        # Note, force will also remove any checking that the keys are in
-        # config.rc_keys
-        with open(rc_path, 'w') as rc:
-            rc.write(yaml.dump(new_rc_config, default_flow_style=False))
-
+    try:
+        json_result = write_config(rc_path, add=args.add, get=args.get,
+            set_=args.set, remove=args.remove, remove_key=args.remove_key, force=args.force)
+    except (NotImplementedError, ConfigValueError, ConfigTypeError, ConfigKeyError) as e:
```

Comment:
I initially thought that we didn't want to catch CouldntParse here, but apparently it is, because it subclasses NotImplementedError. So I guess I'll add it.